openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -124,3 +126,38 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+    def is_saml_user(self, user):
+        """
+        Check and return if the user has an associated SAML social auth record.
+
+        Args:
+            user: django.contrib.auth.models.User instance.
+        Returns:
+            True/False if the user has an associated SAML social auth record.
+        """
+        saml_user = False
+
+        if not user.is_authenticated:
+            return saml_user
+
+        user_social_auth = user.social_auth
```

Comment:
This variable and the following can be merge, however these can be removed if you apply the line 147 suggestion

Reply:
I just wanted to keep the RelatedModel and the query separate, but yes, changed. @andrey-canon